lib/network/internal.nix

```diff
@@ -0,0 +1,220 @@
+{ lib }:
+let
+  inherit (lib)
+    concatStringsSep
+    elemAt
+    fold
+    foldl
+    genList
+    mod
+    range
+    reverseList
+    toInt
+    ;
+
+  inherit (builtins) all any toString;
+in
+rec {
+  common = {
+    /**
+      Given a base and exponent, calculates base raised to the exponent.
+
+      # Example
+
+      ```nix
+      pow 2 3
+      => 8
+      ```
```

Comment:
Note that these internal functions won't have their documentation rendered anywhere, so there's not much point in using markdown style here :)

I'd love to see this kind of documentation in the public interface though!

Reply:
I know but I can't help writing consistent docs for everything :)